packages/core/src/Stores/contract-store.js

```diff
@@ -299,7 +322,33 @@ export default class ContractStore extends BaseStore {
                 );
 
                 main_barrier.updateBarrierShade(true, contract_type);
+
+                barriers = [main_barrier];
+            } else if (isResetContract(contract_type) && entry_spot) {
+                const main_barrier = new ChartBarrierStore(entry_spot, low_barrier, null, {
+                    color: is_dark_mode ? BARRIER_COLORS.DARK_GRAY : BARRIER_COLORS.GRAY,
+                    line_style: BARRIER_LINE_STYLES.SOLID,
+                    not_draggable: true,
+                    hideBarrierLine: false,
+                    shade: DEFAULT_SHADES['2'],
+                });
+
+                main_barrier.updateBarrierShade(true, contract_type);
+
                 barriers = [main_barrier];
+
+                if (reset_time) {
+                    const reset_barrier_object = new ChartBarrierStore(reset_barrier, low_barrier, null, {
+                        color: is_dark_mode ? BARRIER_COLORS.DARK_GRAY : BARRIER_COLORS.GRAY,
+                        line_style: BARRIER_LINE_STYLES.DASHED,
+                        not_draggable: true,
+                        hideBarrierLine: true,
```

Comment:
Hiding barrier (as it was done for Accumulators on line 319), will redraw it in Canvas